lib/modules/manager/flux/extract.ts

```diff
@@ -181,7 +183,24 @@ function resolveResourceManifest(
                 resource.metadata?.namespace)
         );
         if (matchingRepositories.length) {
-          dep.registryUrls = matchingRepositories.map((repo) => repo.spec.url);
+          dep.registryUrls = matchingRepositories
+            .map((repo) => {
+              if (repo.spec.type === 'oci') {
+                // Change datasource to Docker
+                dep.datasource = DockerDatasource.id;
+                // Ensure the URL is a valid OCI path
+                dep.packageName = `${repo.spec.url.replace(
+                  'oci://',
+                  ''
+                )}/${dep.depName!}`;
+                return null;
+              } else if (repo.spec.url.startsWith('oci://')) {
+                return repo.spec.url.replace('oci://', '');
```

Comment:
shouldn't this be same case as before?

Reply:
I don't believe so as I was thinking of handling the case when the type is not set to oci, but the HelmRepository has an OCI url.

However it seems that without the type in HelmRepository, helm datasource would get used, so it might be better to just remove that if statement entirely.